worldengine/draw.py

```diff
@@ -693,7 +696,7 @@ def draw_scatter_plot(world, size, target):
         for x in range(world.width):
             if world.is_land((x, y)):
                 t = world.temperature_at((x, y))
-                p = world.humidity['data'][y, x]
+                p = world.humidity_at((x, y))
```

Comment:
for me it is not about being pretty or not but it is about maintanability through encapsulation: if the latter one was used this patch would be like ten times smaller because we could have changed the way we keep data without affecting the whole codebase. I agree that it could cause a performance hit but in the cases where it is strictly needed we can access directly the array. In many cases we could use numpy functions instead of accessing single values.